File: src/main/java/io/realworld/api/ProfileApi.java
package io.realworld.api;

import io.realworld.model.GetProfileByUsername200Response;

import io.vertx.core.Future;

public interface ProfileApi  {
    Future<ApiResponse<GetProfileByUsername200Response>> followUserByUsername(String username);
    Future<ApiResponse<GetProfileByUsername200Response>> getProfileByUsername(String username);
    Future<ApiResponse<GetProfileByUsername200Response>> unfollowUserByUsername(String username);
}

File: src/main/kotlin/io/realworld/api/ProfileApi.kt
package io.realworld.api

import io.realworld.model.response.ApiResponse
import io.realworld.model.response.GetProfileByUsername200Response
import io.vertx.core.Future

interface ProfileApi {
  fun followUserByUsername(username: String): Future<ApiResponse<GetProfileByUsername200Response>>
  fun getProfileByUsername(username: String): Future<ApiResponse<GetProfileByUsername200Response>>
  fun unfollowUserByUsername(username: String): Future<ApiResponse<GetProfileByUsername200Response>>
}
